contracts/v2/marketplace/TokenMarketplace.sol

```diff
@@ -0,0 +1,344 @@
+
+      // Clear out highest bidder
+      delete offers[_tokenId];
+
+      // Refund it
+      currentHighestBidder.transfer(currentHighestBiddersAmount);
+    }
+  }
+
+  function _handleFunds(uint256 _editionNumber, uint256 _offer, address _currentOwner) internal {
+
+    // Get existing artist commission
+    (address artistAccount, uint256 artistCommissionRate) = kodaAddress.artistCommission(_editionNumber);
+
+    // Get existing optional commission
+    (uint256 optionalCommissionRate, address optionalCommissionRecipient) = kodaAddress.editionOptionalCommission(_editionNumber);
+
+    _splitFunds(artistAccount, artistCommissionRate, optionalCommissionRecipient, optionalCommissionRate, _offer, _currentOwner);
+  }
+
+  function _splitFunds(
+    address _artistAccount,
+    uint256 _artistCommissionRate,
+    address _optionalCommissionRecipient,
+    uint256 _optionalCommissionRate,
+    uint256 _offer,
+    address _currentOwner
+  ) internal {
+
+    // Work out total % of royalties to payout = creator royalties + KO commission
+    uint256 totalCommissionPercentageToPay = defaultKOPercentage.add(defaultArtistRoyaltyPercentage);
+
+    // Send current owner majority share of the offer
+    uint256 totalToSendToOwner = _offer.sub(
+      _offer.div(1000).mul(totalCommissionPercentageToPay)
+    );
+    _currentOwner.transfer(totalToSendToOwner);
+
+    // Send % to KO
+    uint256 koCommission = _offer.div(1000).mul(defaultKOPercentage);
+    koCommissionAccount.transfer(koCommission);
+
+    // Send to seller minus royalties and commission
+    uint256 remainingRoyalties = _offer.sub(koCommission).sub(totalToSendToOwner);
+
+    if (_optionalCommissionRecipient == address(0)) {
+      // After KO and Seller - send the rest to the original artist
+      _artistAccount.transfer(remainingRoyalties);
+    } else {
+      _handleOptionalSplits(_artistAccount, _artistCommissionRate, _optionalCommissionRecipient, _optionalCommissionRate, remainingRoyalties);
+    }
+  }
+
+  function _handleOptionalSplits(
+    address _artistAccount,
+    uint256 _artistCommissionRate,
+    address _optionalCommissionRecipient,
+    uint256 _optionalCommissionRate,
+    uint256 _remainingRoyalties
+  ) internal {
+    uint256 _totalCollaboratorsRate = _artistCommissionRate.add(_optionalCommissionRate);
+    uint256 _scaledUpCommission = _artistCommissionRate.mul(10 ** 18);
+
+    // work out % of royalties total to split e.g. 43 / 85 = 50.5882353%
+    uint256 primaryArtistPercentage = _scaledUpCommission.div(_totalCollaboratorsRate);
+
+    uint256 totalPrimaryRoyaltiesToArtist = _remainingRoyalties.mul(primaryArtistPercentage).div(10 ** 18);
+    _artistAccount.transfer(totalPrimaryRoyaltiesToArtist);
+
+    uint256 remainingRoyaltiesToCollaborator = _remainingRoyalties.sub(totalPrimaryRoyaltiesToArtist);
+    _optionalCommissionRecipient.transfer(remainingRoyaltiesToCollaborator);
+  }
```

Comment:
I am thinking this is sorta binding us only to offer one contributor max - or not allow secondary on multiple collabs (if we ever do them...)

Reply:
Ye, we are bound as KODA v2 only supports two. And this is bound to KODA v2